### DISCUSSION TITLE
Endless Cycle of Recurring Issues

### DISCUSSION THREAD

**Discussion by ohthepain** [fastlane/fastlane#19237] [General]
I keep running into the same issues over and over and don't even know where to start trying to fix fastlane anymore. This is on my Big Sur MacBook and on our M1 build machine, building for iOS and Android.

I have installed using the Gemfile method described here: https://docs.fastlane.tools/getting-started/ios/setup/ and use bundle exec.

I'm not sure what question to ask since I have fixed all of these issues multiple times. It’s like whack-a-mole. Fix a bunch of issues, get a manual build, the next day the cycle continues. I don’t know whether the issue is ruby, gems, ruby version management, or fastlane. I have never seen such madness.

So here are some examples of recurring issues that we have never been able to fix. 

**Example 1: Fastlane keeps reverting to an old version**

# fastlane 2.191.0 is available. You are on 2.134.0.
# You should use the latest version.
Bundle update fastlane
Using fastlane 2.191.0 (was 2.184.1)
Then I can do 
bundle exec fastlane match
> All required keys, certificates and provisioning profiles are installed :raised_hands:

But the next day I have to fix all these issues again, including being back on the old version of fastlane.

**Example 2:** 
Shpretty not found
I have never seen this before today, but it’s just an example

**Example 3:**
I have the issue where I am somehow using an old version that was bundled with fastlane. “To make it easier”. How do I get rid of it? This is apparently because I installed fastlane with brew. Fastlane added their own ruby version “to make it easier”.

**Example 4:**
Could not find dotenv-2.7.6 in any of the sources
[16:38:20]: ▸ Run `bundle install` to install missing gems.
I have run `bundle install` many many times. Running `bundle install` reports no changes.

**Example 5:**
> bundle exec fastlane spaceship
Could not find gem 'pry'
If you installed spaceship using `gem install spaceship` run
 gem install pry
to install the missing gem
> gem install pry

**Example 6:**
Codesign wants to use the fastlane_tmp_codesign-db …

We have never been able to fix any of these issues expect for the shpretty error. We do occasionally manage to get a build out of our build machine. But this is rare and the build machine breaks again within a day or two. I managed to get a build manually today after hours of trying, using 
>  bundle exec fastlane deploy

So I guess my questions are:
What am I missing?
Where to start?
Do I need to install ruby and avoid the system ruby? And if so how? There is a lot of conflicting advice out there.
How do I uninstall this brew-installed fastlane that seems to keep popping up?

**Comment by ohthepain** [comment]
I started by cleaning up ruby. Installed ruby with brew, updated path. I also made sure that fastlane was not installed with brew.

fastlane still keeps going back to 2.184.1. I do a 'bundle update fastlane' every time and then it's back to 2.191.0. I don't know what causes it to revert back to 2.184.1.

**Comment by lucgrabowski** [comment]
@ohthepain After updating bundle and updating to `fastlane (2.191.0)`, did you commit changes in `Gemfile.lock` to your source code repository?

**Reply by ohthepain** [reply]
Yes, I have committed my Gemfile.lock. I was able to resolve the issue. 'Gem info fastlane' reported the two versions of fastlane and I had to remove the older version specifically.

**Reply by ohthepain** [reply]
Now I have this issue https://github.com/fastlane/fastlane/issues/19270 and this one https://github.com/fastlane/fastlane/issues/19264
